plugins/trader/trade.js

```diff
@@ -0,0 +1,293 @@
+
+  The reason I want to pull this out is so that in the future we can have different execution strategies:
+  the current one tries to get the best price - at the cost of execution speed (since it will always
+  make orders it might take a long time before the trade is completed, especially when trying to catch
+  a trend). Once it's split out we can easily create another one that will do a market order instead.
+  Or one that tries to split the order out into multiple small ones (bigger traders need ways to take
+  pressure of the market for example).
+
+*/
+
+/*
+  The Trade class is responsible for overseeing potentially multiple orders
+  to execute a trade that completely moves a position instantiated by the portfolio manager.
+*/
+
+var _ = require('lodash')
+var util = require('../../core/util')
+var dirs = util.dirs()
+var events = require('events')
+var log = require(dirs.core + 'log')
+var async = require('async')
+var checker = require(dirs.core + 'exchangeChecker.js')
+var moment = require('moment')
+
+class Trade{
+  constructor(manager,settings){
+    this.manager = manager
+    this.exchange = manager.exchange
+    this.currency = manager.currency
+    this.asset = manager.asset
+    this.action = settings.action
+    this.isActive = true
+    this.orderIds = []
+
+    log.debug("creating new Trade class to", this.action, this.asset + "/" + this.currency)
+
+    this.doTrade()
+  }
+
+  stop(callback){
+    this.cancelLastOrder(()=>{
+      this.isActive = false
+      log.debug("stopping Trade class from", this.action + "ING", this.asset + "/" + this.currency)
+      if(_.isFunction(callback))
+        callback()
+    })
+  }
+
+  // This function makes sure the limit order gets submitted
+  // to the exchange and initiates order registers watchers.
+  doTrade(retry) {
+    if(!this.isActive)
+      return false
+    
+    // if we are still busy executing the last trade
+    // cancel that one (and ignore results = assume not filled)
+    if(!retry && _.size(this.orderIds))
+      return this.cancelLastOrder(() => this.doTrade());
+
+    var act = function() {
+      var amount, price;
+
+      if(this.action === 'BUY') {
+
+        amount = this.manager.getBalance(this.currency) / this.manager.ticker.ask;
+        if(amount > 0){
+            price = this.manager.ticker.bid;
+            this.buy(amount, price);
+        }
+      } else if(this.action === 'SELL') {
+
+        amount = this.manager.getBalance(this.asset) - this.manager.keepAsset;
+        if(amount > 0){
+            price = this.manager.ticker.ask;
+            this.sell(amount, price);
+        }
+      }
+    };
+    async.series([
+      this.manager.setTicker,
+      this.manager.setPortfolio,
+      this.manager.setFee
+    ], _.bind(act, this));
+
+  };
+
+  // first do a quick check to see whether we can buy
+  // the asset, if so BUY and keep track of the order
+  // (amount is in asset quantity)
+  buy(amount, price) {
+    let minimum = 0;
```

Comment:
pretty sure we had a minimum, right?

Reply:
Yes, that code is just directly pasted from the previous implementation, looks like the minimum is defined later on line `133` before the callback is executed.